webrecorder/browsertrix
# Combine watch crawl with crawl queue

Resolves https://github.com/webrecorder/browsertrix-cloud/issues/601 (see follow-ups for caveats) by integrating the crawl queue and exclusion editor into "Watch Crawls" tab.

### Manual testing
1. Log in and start a crawl
2. Go to Crawl detail page -> Watch Crawl
3. Wait for crawl to start running. Verify crawl queue is shown when watch frames are visible
4. Click "Edit Exclusions". Verify dialog is visible and you're able to add and remove exclusions as expected
5. Close dialog
6. Click "Crawler Instances". Verify dialog is visible and you're able to update scale as expected

### Screenshots
**Combined view:**
<img width="938" alt="Screen Shot 2023-03-16 at 7 28 58 PM" src="https://user-images.githubusercontent.com/4672952/225797507-d8ae385f-3305-4c51-855c-cd35248fc54a.png">

**Combined view loading:**
<img width="940" alt="Screen Shot 2023-03-16 at 7 22 25 PM" src="https://user-images.githubusercontent.com/4672952/225797527-7a639fb4-d6a2-40a4-bb1b-cbb7f6b03906.png">

**Exclusion editor in dialog:**
<img width="1158" alt="Screen Shot 2023-03-16 at 7 21 16 PM" src="https://user-images.githubusercontent.com/4672952/225797636-a3d67c20-c749-461a-b8e3-65d8f07abdc6.png">

**Exclusion editor with pending updates:**
<img width="1150" alt="Screen Shot 2023-03-16 at 7 21 33 PM" src="https://user-images.githubusercontent.com/4672952/225797662-85f6f078-0de5-4995-95e7-73668baaf824.png">

### Follow-ups
As we're actively developing the Workflows rework per [figma note](https://www.figma.com/file/nLjOvnCffAtyCeULXaoVzY/Crawl-Queue-Editing?node-id=0-1&t=CpVQPaI9LKibECfG-0), this is a minimum viable change to be followed by [full-screen editing enhancements](https://github.com/webrecorder/browsertrix-cloud/issues/603) and further [Workflow rework](https://github.com/webrecorder/browsertrix-cloud/issues/652). The functionality of the exclusion editor has not changed in this PR besides reducing the pending exclusions page size to 10 and no longer making that section collapsible.
